src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts

```diff
@@ -574,27 +579,30 @@ export class InstructorSessionsPageComponent extends InstructorSessionModalPageC
           ));
     });
 
-    forkJoin(restoreRequests).subscribe((restoredSessions: FeedbackSession[]) => {
-      restoredSessions.forEach((session: FeedbackSession) => {
-        this.recycleBinFeedbackSessionRowModels = [];
-        const m: SessionsTableRowModel = {
-          feedbackSession: session,
-          responseRate: '',
-          isLoadingResponseRate: false,
-          instructorPrivilege: session.privileges || DEFAULT_INSTRUCTOR_PRIVILEGE,
-        };
-        this.sessionsTableRowModels.push(m);
+    forkJoin(restoreRequests).pipe(finalize(() => this.isRestoreFeedbackSessionLoading = false))
+      .subscribe((restoredSessions: FeedbackSession[]) => {
+        restoredSessions.forEach((session: FeedbackSession) => {
+          this.recycleBinFeedbackSessionRowModels = [];
+          const m: SessionsTableRowModel = {
+            feedbackSession: session,
+            responseRate: '',
+            isLoadingResponseRate: false,
+            instructorPrivilege: session.privileges || DEFAULT_INSTRUCTOR_PRIVILEGE,
+          };
+          this.sessionsTableRowModels.push(m);
+          this.updateInstructorPrivilege(m);
```

Comment:
Hang on, I think this line is no longer necessary, it was removed in #10792

Reply:
I have fixed the requested changes. I apologize for the numerous commits. I had to fix somethings in my workflow, but everything should be good now.